### DISCUSSION TITLE
How to get binding support for Niko 552-721X1 and 552-721X2?

### DISCUSSION THREAD

**Discussion by vyruz1986** [Koenkk/zigbee2mqtt#16727] [Devices]
I recently ordered a Niko 552-721X1 ([Z2M docs link](https://www.zigbee2mqtt.io/devices/552-721X1.html)), and was disappointed to learn it does not support binding to anything. I would really like for these devices to support binding to groups and/or devices, what would be the steps to get this supported? I suppose first the manufacturer needs to be contacted? What would be the correct question to ask them? At this point the best I can come up with is to send them a link to the Z2M docs and ask them if they can add binding support in some firmware upgrade (which Z2M doesn't support for these devices, which is another issue...), but perhaps there is some better terminology I can use?

**Comment by vyruz1986** [comment]
@sjorge I noticed you did much of the initial work in getting this switch supported in https://github.com/Koenkk/zigbee2mqtt/issues/11762.
Could you shed some light on this? I'm very much willing to put in some effort, both by contacting Niko as well as debugging with the switch I have here, but don't really know where to start on this.

**Reply by sjorge** [reply]
Currently the firmware on the device does not support this so there is nothing we from z2m side can do. (Also there are some issues with the X2 variant with button actions that aren't resolved yet).

You could try to contact Niko but I have had very little success with that.

If you do try that route, you can reference the ZCL (https://github.com/Koenkk/zigbee-herdsman/blob/master/docs/07-5123-08-Zigbee-Cluster-Library.pdf)

Starting on 1913 there is reference to the client/server model in this case for bending we want to be able to link something external to the relay and link the build in switch button to a different device.

In decoupled we we can also control the relayed via genOnOff cluster, they just need to allow something other than the coordiantor to bind to it. (Actually  this might already work, I've never tried)

But for the switch button it should be able to bind to genOnOff and maybe genLevelCtrl of a different device.

But as far as I could gather Niko is just relabeling some other device so I am not sure how much they can change wrt the firmware.
I haven't figure out where these are coming from but their [smart plugs](https://www.zigbee2mqtt.io/devices/552-80699.html) are actually rebranded [Leedarson Smart Plugs](https://www.leedarson.com/products/smart-plug)

**Reply by sjorge** [reply]
Hmm actually I wonder if they are just rebranded https://www.leedarson.com/products/eu-on-off-switch
